### ISSUE TITLE
Scheduled/ periodic tasks with celery

### ISSUE THREAD

**Issue by simondrabble** [lorinkoz/django-pgschemas#45]
I have a bunch of scheduled and periodic tasks that currently run under celery. No problem for a single-tenant architecture, but how do I tell celery to run these tasks when it only knows about the public schema? Couple of thoughts:

1. Set up separate celery apps, one for each schema/ tenant. This seems broken by design since now I will have to add code, not just data, for each tenant.
2. Somehow loop through each schema within the celery task processor. This will require writing custom scheduling code, so right off the bat doesn't seem like a good idea, and also.. there's nothing that marks a (postgres) schema as being dynamic, since that's a django-pgschemas thing. So.. cycling through the rows in my Tenants table? Still requires modifying the celery processor itself (celery beat will have to know about schemas).
3. Run a separate celery and beat instance for each tenant. This breaks the whole purpose of multi-tenancy.

To summarize: I want celery beat and other tasks to look at their tables within each pg schema, not just public. 

Anything I'm missing? It looks like django-tenants has a "schema_context" context processor but I don't see that in django-pgschemas.

**Comment by lorinkoz** [comment]
You need to pass the identifier of a tenant to your celery tasks. Then you can use the identifier to retrieve a tenant, then you can wrap your entire task body like this:

```
tenant = Patron.objects.get(....)
with tenant:
    ... # whatever the old task performed
```

**Comment by lorinkoz** [comment]
If you do need to perform the same task for all tenants, then yes, you'll have to do the thing in a loop.

**Comment by lorinkoz** [comment]
@simondrabble Let's please take the discussion to another platform, as I'd rather leave the issues for things to fix. You can reach out to me via email (it's in my profile) and then we can go on via email or agree on something that works for both of us.
